sklearn/ensemble/weight_boosting.py

```diff
@@ -70,16 +69,32 @@ def __init__(self,
         self.learning_rate = learning_rate
         self.random_state = random_state
 
+    def _validate_data(self, X, y=None):
+
+        accept_sparse = ['csr', 'csc']
```

Comment:
It took me a while to understand why we don't just `accept_sparse=True`. Maybe add a comment that these are required for safe_indexing support???

Reply:
`test_sparse_regression` and `test_sparse_classification` check that sparse matrices will be converted in one of the two formats.  Check [here](https://github.com/scikit-learn/scikit-learn/blob/d0c62f60b5455a06a35a462ffe74a9c04306908c/sklearn/ensemble/tests/test_weight_boosting.py#L416) for instance. We could minimize more the validation by turn of these assertions and use `accept_sparse=True`  but `safe_indexing` does not support all sparse matrices. What do you propose?

Reply:
What @jnothman is asking for is simply a comment on the line above saying that these are required for safe_indexing support.